packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+    };
+  }),
+  registerFieldsToInclude: jest.fn((fieldsToInclude) => {
+    return {
+      fieldsToInclude,
+    };
+  }),
+  eventHandler: jest.fn(),
+};
+
+const exampleEngine = {
+  id: 'exampleEngineId',
+  dispatch: functionMocks.dispatch,
+};
+const defaultFieldsToInclude = 'field1,field2';
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  recommendation: 'recommendation',
+  numberOfRecommendations: 10,
+  label: 'Top documents for you',
+  variant: 'grid',
+  recommendationsPerRow: 3,
+  fieldsToInclude: defaultFieldsToInclude,
+};
+
+const selectors = {
+  initializationError: 'c-quantic-component-error',
+  gridContainer: '[data-testid="recommendations__grid"]',
+  carouselContainer: '[data-testid="recommendations__carousel"]',
+  recommendationListItem: '[data-testid="recommendations__item"]',
+  placeholder: 'c-quantic-placeholder',
+  quanticHeading: 'c-quantic-heading',
+};
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticRecommendationList,
+  'c-quantic-recommendation-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildRecommendationList: functionMocks.buildRecommendationList,
+      buildResultTemplatesManager: functionMocks.buildResultTemplatesManager,
+      loadFieldActions: functionMocks.loadFieldActions,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecommendationList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecommendationList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recommendation-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+    console.error = jest.fn();
+  });
+
+  afterEach(() => {
+    cleanup();
+    isInitialized = false;
+    recommendationListState = initialRecommendationListState;
+  });
+
+  describe('when the component is loading', () => {
+    it('should display the placeholder component when loading', async () => {
+      recommendationListState = {
+        isLoading: true,
+        recommendations: [],
+        error: null,
+        searchResponseId: null,
+      };
+      const element = createTestComponent();
+      await flushPromises();
+
+      const loadingPlaceholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+
+      expect(loadingPlaceholder).not.toBeNull();
+    });
+
+    it('should not display the placeholder component when there is an error in the state', async () => {
+      recommendationListState = {
+        isLoading: false,
```

Comment:
The goal is to test when `isLoading` is true but there is an error.
```suggestion
        isLoading: true,
```